### PR TITLE
Feature/task 2

### PR DESCRIPTION
**Pull request by RadekP3**

### CODE REVIEW

File: src/components/List.jsx [hunk 1]
```diff
@@ -1,64 +1,59 @@
...
-        ...item,
-        isInEditMode: false
-      }))
+      items: prevState.items.setIn([itemId, 'isInEditMode'], false),
     }));
   };
 
   _renderListItems = () =>
-    this.state.items.map((item, index) => (
+    this.state.items.valueSeq().map((item, index) => (
```
Comment (kubmir) [inline src/components/List.jsx]: You can improve readability of the code with formatting like this one:
```
this.state.notes
      .valueSeq()
      .map((note, i) => {})
```

File: src/models/ListItem.jsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Record } from 'immutable';
+
+const defaultItem = {
```
Comment (kubmir) [inline src/models/ListItem.jsx]: You named Record as a ListItem. This object describes the default values of the ListItem object. I suggest renaming to the defaultListItem.

File: src/components/List.jsx [hunk 3]
```diff
@@ -1,64 +1,59 @@
...
     }));
   };
 
   _clickLabel = (itemId) => {
     this.setState(prevState => ({
-      items: prevState.items.map(item => (item.id !== itemId ? item : {
-        ...item,
-        isInEditMode: true
-      }))
+      items: prevState.items.setIn([itemId, 'isInEditMode'], true),
```
[the remark this answers is not in the document]
Comment (RadekP3) [reply src/components/List.jsx]: Is it ok if I use mergeIn here? Merge doesn't seem to work for me.

File: src/models/ListItem.js [hunk 4]
```diff
@@ -6,4 +6,4 @@ const defaultListItem = {
   isInEditMode: false
 };
 
-export const ListItem = new Record(defaultListItem);
+export const ListItem = new Record(defaultListItem, 'list item');
```
Comment (vaclavm) [inline src/models/ListItem.js]: The naming convention is to name the record same as the const/file. That would be ListItem. You can fix this into task 3.

**Pull request merged**